### ISSUE TITLE
[eks] [request]: Add/Delete/Update AZs registered with control plane

### ISSUE THREAD

**Issue by sidewinder12s** [aws/containers-roadmap#2258]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Tell us about your request**

Allow users to change the AZs that are configured with an EKS Control Plane. 

Currently you can now modify the subnets that are configured with the control plane but you must provide the same AZs with any changes.

**Which service(s) is this request for?**
EKS

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
I'd really like to just be able to use data lookups for X subnets and create a cluster with it. Instead I end up needing to provide overrides/per cluster configurations on quite a few accounts if we add/remove/change subnet configurations within an account over time. 

Some accounts we've configured VPCs with 4 subnets, then added 2 more. If we don't provide overrides in this case, the EKS change will fail. 

Other accounts we have only 2 AZs configured and we cannot expand it to 3.

In yet another account we have an AZ configured in the VPC which is not supported by EKS so we must filter it out.

**Are you currently working around this issue?**
Maintaining manual overrides

**Attachments**
If you think you might have additional information that you'd like to include via an attachment, please do - we'll take a look. (Remember to remove any personally-identifiable information.)

**Comment by mbarrien** [comment]
Duplicate of https://github.com/aws/containers-roadmap/issues/170

**Comment by peter-nguyen-rw** [comment]
Are there any updates for this feature request? If not, what are workarounds we may do to achieve this without having to destroy an entire EKS cluster and recreate it? @mikestef9

**Comment by peter-nguyen-rw** [comment]
> Are there any updates for this feature request? If not, what are workarounds we may do to achieve this without having to destroy an entire EKS cluster and recreate it? [@mikestef9](https://github.com/mikestef9)

@mikestef9 bump

**Comment by mikestef9** [comment]
This is not on roadmap now. You need to re-create cluster if you need to change AZs from originally passed subnets when creating cluster.

**Comment by coleman-greene-cengage** [comment]
The ability to add additional AZs to an existing EKS cluster should be table stakes. 
The requirement was particularly acute when AWS was throttling availability of instances during the October outage, and the lack of this feature caused customer facing production outages that lasted the majority of that event, as we were unable to scale up.

I've also submitted this as a requirement internally through our TAMs.

**Comment by matt-mercer** [comment]
Adding my support for this,  I can see it might be harder to remove an AZ, but certainly adding a previously unused AZ to the control plane subnets should be supported, imo.

**Comment by nijave** [comment]
> The ability to add additional AZs to an existing EKS cluster should be table stakes. The requirement was particularly acute when AWS was throttling availability of instances during the October outage, and the lack of this feature caused customer facing production outages that lasted the majority of that event, as we were unable to scale up.
> 
> I've also submitted this as a requirement internally through our TAMs.

Agree. Also considering AWS doesn't have the same instance availability in all AZs (even on normal days), being able to keep the CP in the same AZs as the nodes would minimize latency and improve reliability (with CP in `a` and nodes in `b` now either AZ going down breaks the cluster)
